index.d.ts

```diff
@@ -0,0 +1,47 @@
+declare module 'svelte-routing' {
+  import { SvelteComponent, SvelteComponentTyped } from 'svelte'
+  
+  interface RouterProps {
+    basepath?: string
+    url?: string
+  }
+
+  interface RouteProps {
+    path?: string
+    component?: SvelteComponent
+  }
+
+  interface RouteSlots {
+    default: {
+      location: RouteLocation
+      params: RouteParams
+    }
+  }
+
+  interface RouteLocation {
+    pathname: string
+    search: string
+    hash?: string
+    state: {
+      [k in string | number]: unknown
+    }
+  }
+
+  interface RouteParams {
+    [param: string]: string;
+  }
+
+  interface LinkProps {
+    to: string
+    replace?: boolean
+    state?: {}
```

Comment:
Maybe we could extract the definition for the `state` type on line 25 and reuse it here...